cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -596,10 +585,78 @@ instance ToJSON DelegationsAndRewards where
 
 -- Helpers
 
+calcEraInMode
+  :: CardanoEra era
+  -> ConsensusMode mode
+  -> ExceptT ShelleyQueryCmdError IO (EraInMode era mode)
+calcEraInMode era mode=
+  hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (anyCardanoEra era) (AnyConsensusMode mode))
+                   $ toEraInMode era mode
+
+determineEra
+  :: ConsensusModeParams mode
+  -> LocalNodeConnectInfo mode
+  -> ChainPoint
+  -> ExceptT ShelleyQueryCmdError IO AnyCardanoEra
+determineEra cModeParams localNodeConnInfo point =
+  case consensusModeOnly cModeParams of
+    ByronMode -> return $ AnyCardanoEra ByronEra
+    ShelleyMode -> return $ AnyCardanoEra ShelleyEra
+    CardanoMode -> do
+      eraQ <- liftIO . queryNodeLocalState localNodeConnInfo (Just point)
+                     $ QueryCurrentEra CardanoModeIsMultiEra
+      case eraQ of
+        Left acqFail -> left $ ShelleyQueryCmdAcquireFailure acqFail
+        Right anyCarEra -> return anyCarEra
+
+executeQuery
+  :: forall result era mode. CardanoEra era
+  -> ConsensusModeParams mode
+  -> LocalNodeConnectInfo mode
+  -> ChainPoint
+  -> QueryInMode mode (Either EraMismatch result)
+  -> (result -> ExceptT ShelleyQueryCmdError IO ())
+  -- ^ Action to perform on 'result' of 'QueryInMode'
+  -> ExceptT ShelleyQueryCmdError IO ()
+executeQuery era cModeP localNodeConnInfo point q action = do
+  eraInMode <- calcEraInMode era $ consensusModeOnly cModeP
+  case eraInMode of
+    ByronEraInByronMode -> left ShelleyQueryCmdByronEra
+    ShelleyEraInShelleyMode -> do
+      finalRes <- liftIO execQuery
+      handleQueryResult finalRes action
+    ByronEraInCardanoMode ->
+      left ShelleyQueryCmdByronEra
+    ShelleyEraInCardanoMode -> do
+      finalRes <- liftIO execQuery
+      handleQueryResult finalRes action
+    AllegraEraInCardanoMode -> do
+      finalRes <- liftIO execQuery
+      handleQueryResult finalRes action
+    MaryEraInCardanoMode -> do
+      finalRes <- liftIO execQuery
+      handleQueryResult finalRes action
+ where
+   execQuery :: IO (Either AcquireFailure (Either EraMismatch result))
+   execQuery = queryNodeLocalState localNodeConnInfo (Just point) q
```

Comment:
Can we drop the point here too and pass `Nothing`?

Reply:
Same argument here: https://github.com/input-output-hk/cardano-node/pull/2470/files#r595910238